awx/ui_next/src/components/CopyButton/CopyButton.jsx

```diff
@@ -0,0 +1,58 @@
+import React, { useCallback, useEffect } from 'react';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+
+import { Button, Tooltip } from '@patternfly/react-core';
+import { CopyIcon } from '@patternfly/react-icons';
+import useRequest, { useDismissableError } from '@util/useRequest';
+import AlertModal from '@components/AlertModal';
+import ErrorDetail from '@components/ErrorDetail';
+
+function CopyButton({ i18n, itemName, copyItem, disableButtons }) {
+  const {
+    isLoading,
+    error: copyError,
+    request: copyTemplateToAPI,
+  } = useRequest(
+    useCallback(async () => {
+      await copyItem();
+
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []),
+    {}
+  );
+
+  useEffect(() => {
+    if (isLoading) {
+      return disableButtons(true);
+    }
+    return disableButtons(false);
+  }, [isLoading, disableButtons]);
```

Comment:
I'd rename the `disableButtons` prop to `setIsLoading` or something like that. This component doesn’t need to know how the parent components are using that info, simply passing it along.